src/main/java/seedu/nuscents/data/TransactionList.java

```diff
@@ -71,16 +50,16 @@ public void filterTransaction(TransactionCategory category) {
                 filteredTransactions.add(transaction);
                 isFound = true;
             }
+
+            // hacky fix
+            if ((transaction.getCategory() == AllowanceCategory.OTHERS) && (category == ExpenseCategory.OTHERS)) {
+                filteredTransactions.add(transaction);
+                isFound = true;
+            }
```

Comment:
Is this chunk of code to be refactored afterwards?

Reply:
Currently there's this one edge case where filtering by OTHERS only filters expenses and not allowances under the OTHERS category, this is just a quick fix.

Suggestions welcomed!